--- view_send_cli.py	2013-02-12 15:20:26.512219010 -0200
+++ view_send_cli_new.py	2013-02-12 15:09:13.197097606 -0200
@@ -2,19 +2,100 @@
 
 # Form implementation generated from reading ui file './view/send_cli.ui'
 #
-# Created: Tue Feb 12 15:20:26 2013
+# Created: Tue Feb 12 13:46:47 2013
 #      by: PyQt4 UI code generator 4.9.1
 #
 # WARNING! All changes made in this file will be lost!
 
 from PyQt4 import QtCore, QtGui
 
+import devices
+import interfaces
+import os
+
 try:
     _fromUtf8 = QtCore.QString.fromUtf8
 except AttributeError:
     _fromUtf8 = lambda s: s
 
 class Ui_Form(object):
+
+    def isCLI(self, s):
+        if s.find(".cli") == -1:
+            return False
+        else:
+            return True
+
+    def createInterfaceList(self, comboBox=None):
+        if (comboBox==None) :
+            comboBox = self.interfaceComboBox
+        interface_list = interfaces.get_list()
+        for interface in interface_list :
+            comboBox.addItem(_fromUtf8(interface[0]))
+
+    def createDeviceList(self, comboBox=None):
+        if (comboBox==None) :
+            comboBox = self.deviceComboBox
+        device_list = devices.get_list()
+        for device in device_list :
+            comboBox.addItem(_fromUtf8(device))
+
+    def updateMethodList(self, interfaceSelection, listBox=None):
+        if (listBox==None) :
+            listBox = self.methodList
+        listBox.clear()
+        interface_dir = interfaces.get_interface_value(interfaceSelection)
+        files_list = filter(self.isCLI, os.listdir(interface_dir))
+        files_list.append('<none>')
+        for file_name in files_list:
+            item = QtGui.QListWidgetItem()
+            item.setText(_fromUtf8(file_name))
+            font = QtGui.QFont()
+            font.setBold(False)
+            font.setWeight(50)
+            item.setFont(font)
+            self.methodList.addItem(item)
+
+    def updatePreview(self, commandSelection, interfaceSelection=None, previewBox=None):
+        if (interfaceSelection==None) :
+            interfaceSelection = self.interfaceComboBox.currentText()
+        if (previewBox==None) :
+            previewBox = self.previewTextEdit
+        previewBox.clear()
+        if (commandSelection=="") or (commandSelection=="<none>") :
+            return
+        interface_dir = interfaces.get_interface_value(interfaceSelection)
+        command_file = interface_dir+commandSelection
+        command_preview = open(command_file, 'r').read()
+        previewBox.setText(command_preview)
+
+    def acceptSendCLI(self):
+        special_case = False
+        login_param = ""
+        login_flag = ""
+        logout_flag = ""
+        device = str(self.deviceComboBox.currentText())
+        if self.loginCheckBox.isChecked() :
+            login_param = devices.get_device_value(device, "Login")
+            login_flag = "--login"
+            special_case = True
+        if self.logoutCheckBox.isChecked() :
+            logout_flag = "--logout"
+            special_case = True
+        if self.rebootCheckBox.isChecked() :
+            special_case = True
+        device_params = devices.get_device_value(device, "SendCLI")
+        interface_dir = interfaces.get_interface_value(self.interfaceComboBox.currentText())
+        command_file = self.methodList.currentItem().text()
+        arguments = self.argumentsEdit.text()
+        if (command_file=="<none>"):
+                if special_case:
+                    result = "SendCLI_v2.py " + login_param + " " + login_flag + " " + logout_flag + "\n"
+                    print result
+                return
+        result = "SendCLI_v2.py " + device_params + " " + login_param + " -f " + interface_dir + command_file + " " + arguments + " " + login_flag + " " + logout_flag + "\n"
+        print result
+
     def setupUi(self, Form):
         Form.setObjectName(_fromUtf8("Form"))
         Form.resize(401, 388)
@@ -31,8 +112,7 @@
         self.deviceComboBox = QtGui.QComboBox(Form)
         self.deviceComboBox.setAccessibleName(_fromUtf8(""))
         self.deviceComboBox.setObjectName(_fromUtf8("deviceComboBox"))
-        self.deviceComboBox.addItem(_fromUtf8(""))
-        self.deviceComboBox.setItemText(0, _fromUtf8("CPE 1"))
+        self.createDeviceList(self.deviceComboBox)
         self.formLayout.setWidget(0, QtGui.QFormLayout.FieldRole, self.deviceComboBox)
         self.interfaceLabel = QtGui.QLabel(Form)
         self.interfaceLabel.setAccessibleName(_fromUtf8(""))
@@ -41,8 +121,7 @@
         self.interfaceComboBox = QtGui.QComboBox(Form)
         self.interfaceComboBox.setAccessibleName(_fromUtf8(""))
         self.interfaceComboBox.setObjectName(_fromUtf8("interfaceComboBox"))
-        self.interfaceComboBox.addItem(_fromUtf8(""))
-        self.interfaceComboBox.setItemText(0, _fromUtf8("G.SHDSL"))
+        self.createInterfaceList(self.interfaceComboBox)
         self.formLayout.setWidget(1, QtGui.QFormLayout.FieldRole, self.interfaceComboBox)
         self.verticalLayout.addLayout(self.formLayout)
         self.horizontalLayout = QtGui.QHBoxLayout()
@@ -50,13 +129,7 @@
         self.methodList = QtGui.QListWidget(Form)
         self.methodList.setAccessibleName(_fromUtf8(""))
         self.methodList.setObjectName(_fromUtf8("methodList"))
-        item = QtGui.QListWidgetItem()
-        item.setText(_fromUtf8("Methods"))
-        font = QtGui.QFont()
-        font.setBold(False)
-        font.setWeight(50)
-        item.setFont(font)
-        self.methodList.addItem(item)
+        self.updateMethodList(self.interfaceComboBox.currentText(), self.methodList)
         self.horizontalLayout.addWidget(self.methodList)
         self.previewTextEdit = QtGui.QTextEdit(Form)
         self.previewTextEdit.setAccessibleName(_fromUtf8(""))
@@ -101,9 +174,10 @@
 
         self.retranslateUi(Form)
         QtCore.QObject.connect(self.buttonBox, QtCore.SIGNAL(_fromUtf8("rejected()")), Form.close)
-        QtCore.QObject.connect(self.interfaceComboBox, QtCore.SIGNAL(_fromUtf8("currentIndexChanged(QString)")), Form.updateMethodList)
         QtCore.QObject.connect(self.interfaceComboBox, QtCore.SIGNAL(_fromUtf8("currentIndexChanged(QString)")), self.previewTextEdit.clear)
-        QtCore.QObject.connect(self.buttonBox, QtCore.SIGNAL(_fromUtf8("accepted()")), Form.acceptSendCLI)
+        QtCore.QObject.connect(self.interfaceComboBox, QtCore.SIGNAL(_fromUtf8("currentIndexChanged(QString)")), self.updateMethodList)
+        QtCore.QObject.connect(self.methodList, QtCore.SIGNAL(_fromUtf8("currentTextChanged(QString)")), self.updatePreview)
+        QtCore.QObject.connect(self.buttonBox, QtCore.SIGNAL(_fromUtf8("accepted()")), self.acceptSendCLI)
         QtCore.QMetaObject.connectSlotsByName(Form)
 
     def retranslateUi(self, Form):
